inspector/src/components/graph/canvasGraphService.ts

```diff
@@ -126,6 +126,23 @@ export class CanvasGraphService {
         this._attachEventListeners(canvas);
     }
 
+    public resize(size: IPerfLayoutSize) {
+        const { _ctx: ctx } = this;
+        const { width, height } = size;
+        
+        if (!ctx || !ctx.canvas) {
+            return;
+        }
+
+        this._width = width;
+        this._height = height;
+
+        ctx.canvas.width = width;
+        ctx.canvas.height = height;
+
+        this.draw();

```

Comment:
Question: Should I be debouncing draw() inside this method? As it is expensive and resize can be called a lot LOL

Reply:
Ohhhhh Yup good catch, I guess draw should be debouncing ??? so that no matter if called 100 times in a frame, it would handle it gracefully ?

Reply:
Cool latest commit sets up the necessary steps debouncing the draw all the time! it will be called update, and eventually draw will become private (makes more sense conceptually too because external things shouldn't be telling the service when to draw). Going to see if I can make it private at this moment or not because performanceViewerComponent seems to have "draw" right now. Left a to do later note for myself though!

Reply:
Okay replaced draw with _draw, and now we use update() everywhere instead 😄